go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -353,8 +343,135 @@ func getIPAsHexString(ip net.IP) string {
+
+	for _, entry := range n.flowCache {
+		for i, flow := range entry.flows {
+			replacementFlow := flow
+			replacementFlow = strings.Replace(replacementFlow, oldDRIP.String(), n.drIP.String(), ipReplace)
+			replacementFlow = strings.Replace(replacementFlow, portIPRawOld, portIPRawNew, ipReplace)
+
+			replacementFlow = strings.Replace(replacementFlow, oldDRMACRaw, newDRMACRaw, macReplace)
+			replacementFlow = strings.Replace(replacementFlow, oldDRMAC.String(), n.drMAC.String(), macReplace)
+			entry.flows[i] = replacementFlow
+
+		}
+	}
+}
+
+func (n *NodeController) createOrReplaceRoutes(mgmtPortLink netlink.Link, oldDRIP net.IP) error {
+	// hybridOverlay has been initialized and will needs to delete the old routes
+	if oldDRIP != nil {
+		if len(config.HybridOverlay.ClusterSubnets) > 0 {
+			for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+				route := makeRoute(clusterEntry.CIDR, oldDRIP, mgmtPortLink.Attrs().Index)
+				err := util.GetNetLinkOps().RouteDel(route)
+				if err != nil && !os.IsExist(err) {
+					return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+						route.Dst, route.Gw, err)
+				}
+			}
+		} else {
+			nodes, err := n.nodeLister.List(labels.Everything())
+			if err != nil {
+				return err
+			}
+			for _, node := range nodes {
+				if subnet, _ := houtil.ParseHybridOverlayHostSubnet(node); subnet != nil {
+					route := makeRoute(subnet, n.drIP, mgmtPortLink.Attrs().Index)
+					err := util.GetNetLinkOps().RouteDel(route)
+					if err != nil && !os.IsExist(err) {
+						return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+							route.Dst, route.Gw, err)
+					}
+				}
+			}
+		}
+	}
+	// Add a route via the hybrid overlay port IP through the management port
+	// interface for each hybrid overlay cluster subnet
+	if len(config.HybridOverlay.ClusterSubnets) > 0 {
+		for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+			route := makeRoute(clusterEntry.CIDR, n.drIP, mgmtPortLink.Attrs().Index)
+			err := util.GetNetLinkOps().RouteAdd(route)
+			if err != nil && !os.IsExist(err) {
+				return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+					route.Dst, route.Gw, err)
+			}
+		}
+	} else {
+		nodes, err := n.nodeLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		for _, node := range nodes {
+			if subnet, _ := houtil.ParseHybridOverlayHostSubnet(node); subnet != nil {
+				route := makeRoute(subnet, n.drIP, mgmtPortLink.Attrs().Index)
+				err := util.GetNetLinkOps().RouteAdd(route)
+				if err != nil && !os.IsExist(err) {
+					return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+						route.Dst, route.Gw, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
 // EnsureHybridOverlayBridge sets up the hybrid overlay bridge
 func (n *NodeController) EnsureHybridOverlayBridge(node *kapi.Node) error {
+	if atomic.LoadUint32(&n.initialized) == 1 &&
```

Comment:
@JacobTanenbaum is it worth just doing something like:

```
newDRIPStr := node.Annotations[hotypes.HybridOverlayDRIP]
newDRMACStr := node.Annotations[hotypes.HybridOverlayDRMAC]
```
at the top of the function so you don't have to type it out every time below?

Reply:
Also, I'd actually just pull this block out into its own function. I'd rather have it look like:

```
if atomic.LoadUint32(&n.initialized) == 1 {
    if (node.Annotations[hotypes.HybridOverlayDRIP] != n.drIP.String() ||
			node.Annotations[hotypes.HybridOverlayDRMAC] != n.drMAC.String()) {
        if err := n.handleHybridOverlayMACIPChange(node); err != nil {
            return err
        }
    }
    return nil
}
```

Than have two checks for initialized==1...

Reply:
done